src/log/TcpLogger.cc

```diff
@@ -96,6 +96,12 @@ void
 Log::TcpLogger::swanSong()
 {
     disconnect(); // optional: refcounting should close/delete conn eventually
+    if (reconnectScheduled) {
+        // callers must manually manage event queue entries
+        reconnectScheduled->cancel("Log::TcpLogger::swanSong");
+        Events().remove(reconnectScheduled);
```

Comment:
This *is* the public API for class EventScheduler. Remember you vetoed creating a library or namespace scope.

Reply:
The public API is and should be the CancelEvent() or an equivalent stand-alone function. In the current PR code, that function is `eventDelete(AsyncCall::Pointer &c)` declared in `event.h`. Renaming aside, we should use that function here instead of duplicating its internals. I do not understand why you are resisting this seemingly obvious fix.

> Remember you vetoed creating a library or namespace scope.

Neither is required for CancelEvent() to be the public API we use here.

Reply:
Your requests have gone around in circles here back to my original, except using a C library API instead of C++ code. Next step to remove the C-style will be to upgrade the code to C++ syntax again and we have my original code.

Reply:
I see several grave problems with that analysis, but I do not think we need to agree on its validity.

Please let me know if you need clarifications on what to change or would like me to implement the requested changes.